extension-ui/background.js

```diff
@@ -1,84 +1,139 @@
-
 let blockedSitesCache = null;
+let allowedSitesCache = null;
+let isBlackList = true;
 
-// Initialize cache when the extension is loaded
-chrome.runtime.onStartup.addListener(() => initializeBlockedSitesCache());
-chrome.runtime.onInstalled.addListener(() => initializeBlockedSitesCache());
+chrome.runtime.onStartup.addListener(() => initializeCaches());
+chrome.runtime.onInstalled.addListener(() => initializeCaches());
 
-function initializeBlockedSitesCache(callback) {
-  chrome.storage.local.get("blockedSites", (data) => {
+function initializeCaches(callback) {
+  chrome.storage.local.get(["blockedSites", "allowedSites", "isBlackList"], (data) => {
```

Comment:
allowed sites aren't stored in local storage, they are saved in DB
please remove it
you can declare a list in variable for developing time, until we'll get the data from the server